app/code/Magento/Catalog/Ui/DataProvider/Product/ProductDataProvider.php

```diff
@@ -52,12 +60,14 @@ public function __construct(
         array $addFieldStrategies = [],
         array $addFilterStrategies = [],
         array $meta = [],
-        array $data = []
+        array $data = [],
+        PoolInterface $pool = null
     ) {
         parent::__construct($name, $primaryFieldName, $requestFieldName, $meta, $data);
         $this->collection = $collectionFactory->create();
         $this->addFieldStrategies = $addFieldStrategies;
         $this->addFilterStrategies = $addFilterStrategies;
+        $this->pool = $pool ?: ObjectManager::getInstance()->get(PoolInterface::class);
```

Comment:
Do we still need this workaround? As I see, we have declared the argument explicitly in `Magento/Catalog/etc/adminhtml/di.xml`

```xml
<argument name="pool" xsi:type="object">Magento\Catalog\Ui\DataProvider\Product\Listing\Modifier\Pool</argument>
```
So, I don't see a case when we don't have the pool object initialized until its declaration is removed.

@naydav, what do you think?